### ISSUE TITLE
Binary not found error

### ISSUE THREAD

**Issue by leoselig** [AtomLinter/linter-eslint#171]
I know this has come up before but there seems to be no solution 
I am receiving the error

*[Linter-ESLint] eslint binary not found localy, falling back to packaged one. Plugins won't be loaded and linting will possibly not work. (Try Use Global ESLint option, or install localy eslint to your project.)*

When linting a .js/.jsx file. I have eslint locally installed as an npm dependency, deactivated the "use global" option (activating it does not work either when having eslint installed globally) and have an .eslinrc file in my parent directory.

This seems to have also appeared for other Atom linter-* plugins as well. I actually had the same issue  when trying to set up the linter-jsxhint.

Edit: Windows 8.1, Atom 1.0.2, linter-eslint 3.0.0

**Comment by brkattk** [comment]
I'm getting this same issue. I've checked the "Use Global Eslint" option in the plugin's settings, but I'm still seeing this error.

**Comment by c1freitas** [comment]
+1

**Comment by mnquintana** [comment]
IMO this shouldn't even be an error. I think it's okay as a one-time warning notification that linter-eslint is falling back to the packaged binary, but it displays this notification on *every* save *and* as a big red error notification. This should instead be an [info](https://atom.io/docs/api/v1.0.2/NotificationManager#instance-addInfo) or [warning](https://atom.io/docs/api/v1.0.2/NotificationManager#instance-addWarning) notification, and should only be displayed once.

**Comment by gre** [comment]
I also have the error, I don't know why, my projects used to work in previous version of linter-eslint, but since the recent update, it seems to be broken (doesn't resolve the local eslint)

**Comment by gre** [comment]
In Atom Web Console, I have the following error:
```
Cannot find module './lib/testers/eslint-tester'
```

using eslint@1.0.0-rc-2

**Comment by gre** [comment]
Ok my bad, this is actually because of this: https://github.com/eslint/eslint/commit/bf997ef59a621b3f6dc60afee0e5f7d7408441cb

if you guys use latest eslint version, this is why you might have this issue ;)

**Comment by brkattk** [comment]
Wouldn't someone have to manually install the eslint rc version since it's not on npm? My version is `eslint@0.21.2` and I'm getting this error

**Comment by gre** [comment]
actually it is published on NPM but this is not visible in npmjs.org interface.

```
npm install eslint@1.0.0-rc-2
```

**Comment by gre** [comment]
Have you tried to open the Atom Web Console to see if there is a more explicit error?

**Comment by brkattk** [comment]
Strange. I had deactivated the package yesterday since it kept throwing the error. I haven't updated anything since then. Now I just enabled the package again to look in the console and it appears to work just fine.

**Comment by ismyrnow** [comment]
It seems like the issue might be that it doesn't pickup the "use global" setting immediately.

I found the errors went away after adding a global node path, checked use global, and then disabled/enabled the plugin.

**Comment by dcrockwell** [comment]
+1

**Comment by dcrockwell** [comment]
I was only having this issue on a single repo, so I looked at the difference between the two in package.json:

* the repos with eslint version `^0.24.0` find the binary and work properly
* the repos with eslint version `^1.0.0` **do not** find the binary

I switched the problematic repo dependency to eslint `^0.24.0` and everything worked.

I also have `babel-eslint ^4.0.5` installed. Not sure if that makes any difference.

**Comment by aknuds1** [comment]
I'm also having this problem, having enabled global ESLint:

**~/.atom/config.cson**:

    'linter-eslint':
      'useGlobalEslint': true

Why is it taking so long to fix I wonder?

**Comment by leoselig** [comment]
I am not sure if this has been fixed somehow.

However, after watching the package's behavior carefully over the past months, I can safely say that the error now only occurs in my described setup if actually I forgot to install eslint locally. Installing it fixes the message as it should.

Since the error did not come up again for some time now the issue could be closed from my side. 

Then again, I don't know what happens on the machines of the other users here.

**Comment by steelbrain** [comment]
Fixed by v5.0.0
